yew/src/virtual_dom/vlist.rs

```diff
@@ -94,77 +94,211 @@ impl VDiff for VList {
+            self.props.inner.clone()
         }
     }
 
     #[test]
-    fn check_fragments() {
-        let fragment = html! {
-            <>
-            </>
-        };
-        html! {
-            <div>
-                { fragment }
-            </div>
-        };
+    fn vlist_vdiff_apply_non_keyed_from_none_works_with_all_vnode_types_as_children() {
+        let scheduler = yew::scheduler::scheduler();
+
+        let vchild_with_tag: VChild<Comp> = VChild::new(
+            CompProps {
+                inner: html! { <p>{0}</p> },
+            },
+            NodeRef::default(),
+            None,
+        );
+
+        let vchild_with_list: VChild<Comp> = VChild::new(
+            CompProps {
+                inner: html! { <>{"(list)"}</> },
+            },
+            NodeRef::default(),
+            None,
+        );
+
+        let vchild_with_comp: VChild<Comp> = VChild::new(
+            CompProps {
+                inner: html! { <Comp inner=html!{ "COMP" } /> },
+            },
+            NodeRef::default(),
+            None,
+        );
+
+        let element: Element = crate::utils::document().create_element("br").unwrap();
+        let vchild_with_ref: VChild<Comp> = VChild::new(
+            CompProps {
+                inner: html! { <Comp inner=VNode::VRef(element.clone().into()) /> },
+            },
+            NodeRef::default(),
+            None,
+        );
+
+        let vchild_empty: VChild<Comp> =
+            VChild::new(CompProps { inner: html! {} }, NodeRef::default(), None);
+
+        let vref_element: Element = crate::utils::document().create_element("i").unwrap();
+        let vref_node: Node = vref_element.clone().into();
+        let mut vtag = VTag::new("span");
+        vtag.children.add_child(VTag::new("hr").into());
+        let mut vlist = VList::new_with_children(
+            vec![
+                VNode::VText(VText::new("a".into())),
+                VNode::VTag(Box::new(vtag)),
+                VNode::VComp(vchild_with_ref.into()),
+                VNode::VText(VText::new("c".into())),
+                VNode::VComp(vchild_with_list.into()),
+                VNode::VText(VText::new("d".into())),
+                VNode::VComp(vchild_with_tag.into()),
+                VNode::VComp(vchild_empty.into()),
+                VNode::VList(VList::new_with_children(
+                    vec![
+                        VNode::VText(VText::new("foo".into())),
+                        VNode::VComp(vchild_with_comp.into()),
+                        VNode::VText(VText::new("bar".into())),
+                    ],
+                    None,
+                )),
+                VNode::VRef(vref_node),
+            ],
+            None,
+        );
+
+        let mut vlist_copy = vlist.clone();
+        let parent_scope: AnyScope = Scope::<Comp>::new(None).into();
+        let parent_element = crate::utils::document().create_element("div").unwrap();
+
+        let scheduler_lock = scheduler.lock();
```

Comment:
Could you add a comment to explain why this lock is needed?

Reply:
Wasn't needed, removed!